pyproject.toml

```diff
@@ -3,3 +3,10 @@ profile="jinja"
 linter_output_format = "{filename}:{line}: {code} {message} {match}"
 # Ignore linting rules that are related to web pages
 ignore = "H016,H021,H030,H031"
+
+[tool.ruff]
+extend-exclude = ["data", "models", "db", "cache"]
+output-format = "grouped"
+
+[tool.ruff.lint]
+select = ["E4", "E7", "E9", "F", "I"]
```

Comment:
The Ruff docs say:
> # Enable Pyflakes (`F`) and a subset of the pycodestyle (`E`)  codes by default.
> # Unlike Flake8, Ruff doesn't enable pycodestyle warnings (`W`) or
> # McCabe complexity (`C901`) by default.

Given we were using flake8, let's enable pycodestyle warnings and McCabe complexity too

Reply:
It seems that `C901` is not passing, which indicates that it was not enabled. Also, our `flake8` config was ignoring most of the `W` rules, so we may not want to enable them, which will be closer to the defaults.